lib/api/api-request.js

```diff
@@ -52,6 +61,10 @@ class RequestHandler extends AsyncResource {
     this.context = null
     this.onInfo = onInfo || null
 
+    if (channels.trailers.hasSubscribers) {
+      this.bodyReceived = []
+    }
```

Comment:
Unfortunately keeping all the body in memory for all the users of the trailers event is significantly too expensive. Why can't you pass the `this.res` object to a previous event?

Reply:
If you consume the body, it can't be consumed later on (I have converted it to a Readable stream, just haven't pushed it yet). Ie:

```js
import diagnosticsChannel from 'diagnostics_channel'
import { request } from 'undici'
import { text } from 'stream/consumers'

diagnosticsChannel.channel('undici:request:trailers').subscribe(({ response }) => {
    // let's say it was the same readable stream
    // consume the body somehow
    text(response.body).then(...)
    // will likely throw because the body has been used already
})

const { body } = await reques('...')
await body.text() // do something with body after
```

Reply:
How about?

```js

import diagnosticsChannel from 'diagnostics_channel'
import { request } from 'undici'
import { text } from 'stream/consumers'

diagnosticsChannel.channel('undici:request:trailers').subscribe(({ response }) => {
   const push = response.push
   response.push = (chunk) => {
     // do whatever you want here
     push.call(response, chunk)
   }
})

const { body } = await reques('...')
await body.text() // do something with body after

```

Reply:
You receive the diagnostics_channel message after the body has been consumed (in this example and others I tested), so we only get a chunk that is null.

Reply:
After looking into this, we can get the body - as an array of buffers, on the main branch - using:

```js
import diagnosticsChannel from 'diagnostics_channel'
import { request } from './index.js'

diagnosticsChannel.channel('undici:request:trailers').subscribe(({ response }) => {
  const kConsume = Object.getOwnPropertySymbols(response).find(s => s.description === 'kConsume');
  console.log(response[kConsume].body);
})

const { body } = await request('https://www.google.com/')
await body.text() // do something with body after

```

Reply:
```js
let body1 = '', body2 = ''
diagnosticsChannel.channel('undici:request:trailers').subscribe(({ response }) => {
  const kConsume = Object.getOwnPropertySymbols(response).find(s => s.description === 'kConsume');
  body2 = Buffer.concat(response[kConsume].body).toString();
})

const { body } = await request('https://www.google.com/')
body1 = await body.text() // do something with body after

console.log(body1 === body2); // true
```

The bodies are the same 🤯

Reply:
I have 0 idea how this property works 😅. When fetching stuff on localhost, this property doesn't exist, but when fetching any external server, it's there.. This is making writing tests pretty difficult.